lms/services/canvas.py

```diff
@@ -32,7 +32,12 @@ def public_url_for_file(
 
         try:
             if check_in_course:
-                self._course_copy_plugin.is_file_in_course(course_id, effective_file_id)
+                if not self._course_copy_plugin.is_file_in_course(
+                    course_id, effective_file_id
+                ):
+                    raise FileNotFoundInCourse(
```

Comment:
D2L and blackboard implementations do the raise here in the service, doing the same here.